packages/client/src/components/ClientProvider/ClientProvider.tsx

```diff
@@ -161,7 +159,13 @@ const ClientProvider = ({
               error: CombinedError,
               operation: Operation<unknown, AnyVariables>,
             ) => {
-              const errorMessages = extractErrorMessages(error);
+              let errorMessages = extractErrorMessages(error);
+
+              const validationMessageKeys = extractValidationMessageKeys(error);
+              if (validationMessageKeys) {
```

Comment:
This will return an array always, either with error enums or an empty array, in the case of the latter I do not think we want to replace errorMessages otherwise we lose things like error messages due to authorization

```suggestion
              if (validationMessageKeys.length > 0) {
```